cpp/tests/utilities/test_utilities_impl.cuh

```diff
@@ -59,5 +60,109 @@ graph_to_host_coo(
+  }
+
+  rmm::device_uvector<edge_t> d_offsets(0, handle.get_stream());
+
+  if (d_wgt) {
+    std::tie(d_offsets, d_dst, *d_wgt, std::ignore) =
+      detail::compress_edgelist<edge_t, store_transposed>(d_src.begin(),
+                                                          d_src.end(),
+                                                          d_dst.begin(),
+                                                          d_wgt->begin(),
+                                                          vertex_t{0},
+                                                          std::optional<vertex_t>{std::nullopt},
+                                                          graph_view.number_of_vertices(),
+                                                          vertex_t{0},
+                                                          graph_view.number_of_vertices(),
+                                                          handle.get_stream());
+
+    // segmented sort neighbors
+    detail::sort_adjacency_list(handle,
+                                raft::device_span<edge_t const>(d_offsets.data(), d_offsets.size()),
+                                d_dst.begin(),
+                                d_dst.end(),
+                                d_wgt->begin());
+  } else {
+    std::tie(d_offsets, d_dst, std::ignore) =
+      detail::compress_edgelist<edge_t, store_transposed>(d_src.begin(),
+                                                          d_src.end(),
+                                                          d_dst.begin(),
+                                                          vertex_t{0},
+                                                          std::optional<vertex_t>{std::nullopt},
+                                                          graph_view.number_of_vertices(),
+                                                          vertex_t{0},
+                                                          graph_view.number_of_vertices(),
+                                                          handle.get_stream());
+    // segmented sort neighbors
+    detail::sort_adjacency_list(handle,
+                                raft::device_span<edge_t const>(d_offsets.data(), d_offsets.size()),
+                                d_dst.begin(),
+                                d_dst.end());
+  }
+
+  return std::make_tuple(
+    to_host(handle, raft::device_span<edge_t const>(d_offsets.data(), d_offsets.size())),
+    to_host(handle, raft::device_span<vertex_t const>(d_dst.data(), d_dst.size())),
+    d_wgt ? to_host(handle, raft::device_span<weight_t const>(d_wgt->data(), d_wgt->size()))
+          : std::optional<std::vector<weight_t>>(std::nullopt));
+}
+
+template <typename vertex_t, typename edge_t, typename weight_t, bool store_transposed>
+std::tuple<cugraph::graph_t<vertex_t, edge_t, weight_t, store_transposed, false>,
+           std::optional<rmm::device_uvector<vertex_t>>>
+mg_graph_to_sg_graph(
```

Comment:
I think this is a better mechanism than reading/creating a separate SG graph as we have been doing elsewhere (For tests with R-mat graphs, we rely on the fact that our pseduo random number generator is deterministic for a given seed, but we no longer need to rely on this).

Not a high priority thing, but we may better apply this mechanism in all the other MG C++ tests as well.

Reply:
I was thinking this when I created it.  Once we merge this PR I'll add an issue to the backlog to update the other tests.